proto/adaptors/0x/zero_x.proto

```diff
@@ -0,0 +1,38 @@
+// Protos for calling functions on the 0x adapter.
+
+syntax = "proto3";
+package steward.v3;
+
+option go_package = "/steward_proto";
+
+import "adaptors/base.proto";
+
+message ZeroXAdaptorV1 {
+    oneof function {
+        /***** BASE ADAPTOR FUNCTIONS *****/
+
+        // Represents function `revokeApproval(ERC20 asset, address spender)`
+        RevokeApproval revoke_approval = 1;
+
+        /***** ADAPTOR-SPECIFIC FUNCTIONS *****/
+
+        // Represents function `swapWith0x(ERC20 tokenIn, ERC20 tokenOut, uint256 amount, bytes memory swapCallData)`
+        SwapWith0x swap_with_0x = 2;
+    }
+
+    /*
+    * Allows strategists to make ERC20 swaps using 0x.
+    *
+    * Represents function `swapWith0x(ERC20 tokenIn, ERC20 tokenOut, uint256 amount, bytes memory swapCallData)`
+    */
+    message SwapWith0x {
+        string token_in = 1;
+        string token_out = 2;
+        string amount = 3;
+        bytes swap_call_data = 4;
```

Comment:
The respective platforms have a service that they will get the encoded bytes from and pass them in. There was a question about whether to make them encode it as raw bytes (since they get back a hex string) or have steward do it, I was going to have a look at doing it with steward and make these fields strings.